Patch ported from Debian
http://anonscm.debian.org/cgit/pkg-gnupg/libgpg-error.git/diff/?id=c3d7571

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>


From 91da4f5dbbc9d93975ef9753652a4e71719f9f27 Mon Sep 17 00:00:00 2001
From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Mon, 16 Mar 2015 13:26:00 -0400
Subject: [LIBGPG-ERROR PATCH] avoid breakage with gcc 5
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

 * src/Makefile.am: add -P to the C preprocessor when building
   mkerrcodes.h, to avoid a noisy intermediate pipeline.

--

With gcc 5 without this patch, we see many errors like the following:

gcc -I. -I. -o mkerrcodes ./mkerrcodes.c
In file included from ./mkerrcodes.c:26:0:
./mkerrcodes.h:9:5: error: expected expression before ‘,’ token
   { , "GPG_ERR_E2BIG" },
     ^
./mkerrcodes.h:10:5: error: expected expression before ‘,’ token
   { , "GPG_ERR_EACCES" },
     ^

This patch cleans up the generated mkerrcodes.h by making the
intermediate stage clean for all the versions of gcc i tested (4.x and
5).

Debian-Bug-Id: 777374
Signed-Off-By: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
---
 src/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 99c2c53..f847a80 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -213,7 +213,7 @@ code-to-errno.h: Makefile mkerrnos.awk errnos.in
 # It is correct to use $(CPP).  We want the host's idea of the error codes.
 mkerrcodes.h: Makefile mkerrcodes.awk $(gpg_extra_headers)
 	$(AWK) -f $(srcdir)/mkerrcodes1.awk $(srcdir)/errnos.in >_$@
-	$(CPP) $(CPPFLAGS) $(extra_cppflags) _$@ | grep GPG_ERR_ | \
+	$(CPP) $(CPPFLAGS) $(extra_cppflags) -P _$@ | grep GPG_ERR_ | \
                $(AWK) -f $(srcdir)/mkerrcodes.awk >$@
 	-rm _$@
 
-- 
2.1.4

